### PR TITLE
update(docs): improve onboarding experience

### LINKED ISSUE #33: Improve README or CONTRIBUTING docs for new contributors
~~Add a link to MongoDB installation instructions.~~
~~Add details about API keys.~~
~~Add information about how to run the test server locally.~~
~~Add details on how to run the unit tests.~~

### LINKED ISSUE #68: Use npm-scripts to make onboarding easier
- Use a postinstall script to install `grunt-cli` and `bower`.
- Add a prestart script to build before serving.
- Update README to be more simple.

### PR DESCRIPTION
**Pull request by Splaktar**
Added details for creating Google API key, clientId, and clientSecret to README.
Add wiki link to ingesting data into local Hub instance to README.
fix JSHint failures
fix 'gloal' typo in monthly_metrics recording
add bower and grunt-cli to `package.json`. Update to node 0.12.0 minimum.
Add `postinstall` and `prestart` scripts for `npm`. Updated README with new npm commands.

Closes #68. Closes #33.

### CODE REVIEW

File: lib/controllers/auth.js [hunk 1]
```diff
@@ -77,7 +79,7 @@ module.exports = {
 							});
 						} else {
 							req.logout();
-							res.send(403,"Unauthorized");
+							res.send(403,'Unauthorized');
```
Comment (tasomaniac) [inline lib/controllers/auth.js]: space after `,` ? :smile:

File: lib/utils.js [hunk 2]
```diff
@@ -115,7 +119,7 @@ module.exports = {
 
 			var saveMetric = function() {
 				var set = {};
-				set["values."+month] = value;
+				set['values.'+month] = value;
```
Comment (tasomaniac) [inline lib/utils.js]: Space before and after `+` ?

**Comment by tasomaniac** [comment]
Wow! That was a big PR. :) I went over all the lines. I think it looks good. I think this should be reviewed by 2 people in total. So let's wait for another one before merging it.

File: lib/controllers/auth.js (hunk 1 above)
Comment (Splaktar) [inline lib/controllers/auth.js]: Done

File: lib/utils.js (hunk 2 above)
Comment (Splaktar) [inline lib/utils.js]: Done

**Comment by Splaktar** [comment]
@tasomaniac Thank you very much. I can't wait too long because I need to prepare for my GDG-X presentation in Brazil this week and I want this on boarding experience to be tried by a couple more people before then.

**Comment by tasomaniac** [comment]
That makes sense. I merged it.

**Pull request merged**